applications/plugins/Geomagic/src/GeomagicDriver.cpp

```diff
@@ -254,10 +260,22 @@ void GeomagicDriver::initDevice()
     m_hHD = hdInitDevice(d_deviceName.getValue().c_str());
 
     if (HD_DEVICE_ERROR(error = hdGetError()))
-    {
-        msg_error() << "Failed to initialize the device called " << d_deviceName.getValue().c_str();
-        d_omniVisu.setValue(false);
+    {        
         m_errorDevice = error.errorCode;
+        if (m_errorDevice == 769) // double initialisation, will try to close and reinit device
+        {
+            msg_warning() << "Device has already been initialized. Will clear driver and reinit the device properly.";
+            m_hHD = hdGetCurrentDevice();
```

Comment:
Are you sure you want to erase the value of `m_hHD` which was set a few lines before?

Reply:
yes because I think (90% sure) that the hdInitDevice doesn't return a valid m_hHD id when init failed. Here is failed because tool is already init a waiting so we need to get the current device id.